client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -203,8 +214,41 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
-        Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
-        synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        while (true) {
+            synchronizer.fetchUpdates();
```

Comment:
The update function in stateTransition will do this.

Reply:
The while loop in `resetReaderGroup` is so that we can retry in case the stateTransition fails.

Reply:
I think you misunderstand:
Calling `update` on a state synchronizer will be retried can cause the update function to be re-run if it is not up to date. So it is not necessary to call fetchUpdates between calls to `update` in a loop because by definition the state will be up to date. That is the contract of the `update` function.

Reply:
I've moved the `synchronizer.fetchUpdates` to outside the while loop so its called once before the while loop. Then within the while loop `update` will allow for the state to be up to date.